tests/integration/e2e/MultiYearPlan.test.tsx

```diff
@@ -0,0 +1,183 @@
+      imports: [
+        SessionModule.forRoot({
+          session: {
+            secret: dummy.safeString,
+            resave: true,
+            saveUninitialized: true,
+          },
+        }),
+        ConfigModule,
+        LogModule,
+        TypeOrmModule.forRootAsync({
+          imports: [ConfigModule],
+          useFactory: (
+            config: ConfigService
+          ): TypeOrmModuleOptions => ({
+            ...config.dbOptions,
+            synchronize: true,
+            autoLoadEntities: true,
+            retryAttempts: 10,
+            retryDelay: 10000,
+          }),
+          inject: [ConfigService],
+        }),
+        AuthModule.register({
+          strategies: [TestingStrategy],
+          defaultStrategy: AUTH_MODE.TEST,
+        }),
+        PopulationModule,
+        CourseModule,
+        FacultyModule,
+        CourseInstanceModule,
+        MetadataModule,
+        SemesterModule,
+      ],
+      controllers: [
+        UserController,
+      ],
+    })
+      .overrideProvider(ConfigService)
+      .useValue(fakeConfig)
+      .compile();
+    courseRepository = testModule.get(
+      getRepositoryToken(Course)
+    );
+    const nestApp = await testModule
+      .createNestApplication()
+      .useGlobalPipes(new BadRequestExceptionPipe())
+      .init();
+    const api = nestApp.getHttpServer();
+    request.defaults.adapter = mockAdapter(api);
+    authStub.resolves(dummy.adminUser);
+  });
+  afterEach(async function () {
+    return testModule.close();
+  });
+  describe('Rendering Instructor Information', function () {
+    let parentCourse: Course;
+    let childCourse: Course;
+    beforeEach(async function () {
+      parentCourse = await courseRepository.findOne();
+      childCourse = await courseRepository.findOne({
+        where: {
+          id: Not(parentCourse.id),
+        },
+      });
+      renderResult = render(
+        <MemoryRouter initialEntries={['/course-admin']}>
+          <App />
+        </MemoryRouter>
+      );
+    });
+    context('when a course is a child of another', function () {
+      it('should display the faculty of its parent course', async function () {
+        await renderResult.findByText(childCourse.title, { exact: false });
+        // Edit the child course to set its parent via the sameAs field
+        const editButton = await renderResult
+          .findByLabelText(`Edit course information for ${childCourse.title}`,
+            { exact: false });
+        fireEvent.click(editButton);
+        await renderResult.findByRole('dialog');
+        const sameAsSelect = renderResult.getByLabelText('Same As', { exact: false }) as HTMLSelectElement;
+        fireEvent.change(
+          sameAsSelect,
+          { target: { value: parentCourse.id } }
+        );
+        const submitButton = renderResult.getByText('Submit');
+        fireEvent.click(submitButton);
+        await wait(() => !renderResult.queryByText('required fields', { exact: false }));
```

Comment:
I've also been seeing the same issue here that I noted in the `SchedulePage` test, where the tests are failing because it seems to be navigating before the data is completely saved.

Reply:
This was also updated in commit [18c4db7](https://github.com/seas-computing/course-planner/pull/652/commits/18c4db7647edc1d66c9425105001dd4afa479d77).